backend/src/schema/resolvers/searches.spec.js

```diff
@@ -0,0 +1,444 @@
+    findResources(query: $query, limit: 5) {
+      __typename
+      ... on Post {
+        id
+        title
+        content
+      }
+      ... on User {
+        id
+        slug
+        name
+      }
+    }
+  }
+`
+describe('resolvers/searches', () => {
+  let variables
+
+  describe('given one user', () => {
+    beforeAll(async () => {
+      user = await Factory.build('user', {
+        id: 'a-user',
+        name: 'John Doe',
+        slug: 'john-doe',
+      })
+      authenticatedUser = await user.toJson()
+    })
+
+    describe('query contains first name of user', () => {
+      it('finds the user', async () => {
+        variables = { query: 'John' }
+        await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+          data: {
+            findResources: [
+              {
+                id: 'a-user',
+                name: 'John Doe',
+                slug: 'john-doe',
+              },
+            ],
+          },
+        })
+      })
+    })
+
+    describe('adding one post', () => {
+      beforeAll(async () => {
+        await Factory.build(
+          'post',
+          {
+            id: 'a-post',
+            title: 'Beitrag',
+            content: 'Ein erster Beitrag',
+          },
+          { authorId: 'a-user' },
+        )
+      })
+
+      describe('query contains title of post', () => {
+        it('finds the post', async () => {
+          variables = { query: 'beitrag' }
+          await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+            data: {
+              findResources: [
+                {
+                  __typename: 'Post',
+                  id: 'a-post',
+                  title: 'Beitrag',
+                  content: 'Ein erster Beitrag',
+                },
+              ],
+            },
+          })
+        })
+      })
+
+      describe('casing', () => {
+        it('does not matter', async () => {
+          variables = { query: 'BEITRAG' }
+          await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+            data: {
+              findResources: [
+                {
+                  __typename: 'Post',
+                  id: 'a-post',
+                  title: 'Beitrag',
+                  content: 'Ein erster Beitrag',
+                },
+              ],
+            },
+          })
+        })
+      })
+
+      describe('query consists of words not present in the corpus', () => {
+        it('returns empty search results', async () => {
+          await expect(
+            query({ query: searchQuery, variables: { query: 'Unfug' } }),
+          ).resolves.toMatchObject({ data: { findResources: [] } })
+        })
+      })
+
+      describe('testing different post content', () => {
+        beforeAll(async () => {
+          return Promise.all([
+            Factory.build(
+              'post',
+              {
+                id: 'b-post',
+                title: 'Aufruf',
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-17T09:37:17Z" title="Tuesday, March 17th 2020, 10:37:17 am +01:00">Mar 17, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,452 @@
+import Factory, { cleanDatabase } from '../../db/factories'
+import { gql } from '../../helpers/jest'
+import { getNeode, getDriver } from '../../db/neo4j'
+import createServer from '../../server'
+import { createTestClient } from 'apollo-server-testing'
+
+let query, authenticatedUser
+
+const driver = getDriver()
+const neode = getNeode()
+
+beforeAll(async () => {
+  await cleanDatabase()
+  const { server } = createServer({
+    context: () => {
+      return {
+        driver,
+        neode,
+        user: authenticatedUser,
+      }
+    },
+  })
+  query = createTestClient(server).query
+})
+
+afterAll(async () => {
+  await cleanDatabase()
+})
+
+const searchQuery = gql`
+  query($query: String!) {
+    findResources(query: $query, limit: 5) {
+      __typename
+      ... on Post {
+        id
+        title
+        content
+      }
+      ... on User {
+        id
+        slug
+        name
+      }
+    }
+  }
+`
+let user
+
+describe('resolvers', () => {
+  describe('searches', () => {
+    let variables
+
+    describe('given one user', () => {
+      beforeAll(async () => {
+        user = await Factory.build('user', {
+          id: 'a-user',
+          name: 'John Doe',
+          slug: 'john-doe',
+        })
+        authenticatedUser = await user.toJson()
+      })
+
+      const factoryOptions = {
+        authorId: 'a-user',
+      }
+
+      describe('query contains first name of user', () => {
+        it('finds the user', async () => {
+          variables = { query: 'John' }
+          await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+            data: {
+              findResources: [
+                {
+                  id: 'a-user',
+                  name: 'John Doe',
+                  slug: 'john-doe',
+                },
+              ],
+            },
+          })
+        })
+      })
+
+      describe('adding one post', () => {
+        beforeAll(async () => {
+          await Factory.build(
+            'post',
+            {
+              id: 'a-post',
+              title: 'Beitrag',
+              content: 'Ein erster Beitrag',
+            },
+            factoryOptions,
+          )
+        })
+
+        describe('query contains title of post', () => {
+          it('finds the post', async () => {
+            variables = { query: 'beitrag' }
+            await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+              data: {
+                findResources: [
+                  {
+                    __typename: 'Post',
+                    id: 'a-post',
+                    title: 'Beitrag',
+                    content: 'Ein erster Beitrag',
+                  },
+                ],
+              },
+            })
+          })
+        })
+
+        describe('casing', () => {
+          it('does not matter', async () => {
+            variables = { query: 'BEITRAG' }
+            await expect(query({ query: searchQuery, variables })).resolves.toMatchObject({
+              data: {
+                findResources: [
+                  {
+                    __typename: 'Post',
+                    id: 'a-post',
+                    title: 'Beitrag',
+                    content: 'Ein erster Beitrag',
+                  },
+                ],
+              },
+            })
+          })
+        })
+
+        describe('query consists of words not present in the corpus', () => {
+          it('returns empty search results', async () => {
+            await expect(
+              query({ query: searchQuery, variables: { query: 'Unfug' } }),
+            ).resolves.toMatchObject({ data: { findResources: [] } })
+          })
+        })
+
+        describe('testing different post content', () => {
```


this is a good use of a describe block, since the set up is different. Although, it's not immediately clear to me why these posts could be created in a up the stream describe block and then tested in `it` blocks further down...
maybe you wanted to have the `build` closer to the test to make it easier to see the specifics of each post?

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-17T11:03:13Z" title="Tuesday, March 17th 2020, 12:03:13 pm +01:00">Mar 17, 2020</time>_

---


I need async calls to the factory. describe cannot be used async. This would leed to a beforeAll call before each new test and therefore a even deeper nesting of describe blocks. So I decided to write all the data to the DB firsrt, and then explain each test case later.